### PR TITLE
common: drop the unused zero-copy facilities in ceph::bufferlist

### PR DESCRIPTION
**Pull request by rzarzynski**
These commits remove the kernel-assisted (with `tee`, `splice` and potentially `vmsplice`) zero copy facilities we have in `ceph::buffer` but nowhere use. Except making things easier, the benefit comes from killing the dynamic polymorphism overhead over accessing the `data` field of `buffer::raw` implementations.

### CODE REVIEW

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-09-12_06:54:12-rados-wip-kefu-testing-2018-09-12-1250-distro-basic-smithi/

**Comment by tchaikov** [comment]
retest this please.

File: src/include/buffer_raw.h [hunk 1]
```diff
@@ -77,7 +77,7 @@ namespace ceph::buffer {
     raw(const raw &other) = delete;
     const raw& operator=(const raw &other) = delete;
 public:
-    virtual char *get_data() {
+    virtual char *get_data() final {
```
[the remark this answers is not in the document]
Comment (rzarzynski) [reply src/include/buffer_raw.h]: Dropped `virtual` entirely after the run (thanks, @tchaikov).

**Comment by rzarzynski** [comment]
jenkins retest this please

**Pull request merged**